src/System.Linq/src/System/Linq/Take.cs

```diff
@@ -21,14 +21,12 @@ public static IEnumerable<TSource> Take<TSource>(this IEnumerable<TSource> sourc
                 return EmptyPartition<TSource>.Instance;
             }
 
-            IPartition<TSource> partition = source as IPartition<TSource>;
-            if (partition != null)
+            if (source is IPartition<TSource> partition)
```

Comment:
💭 Seems like some of these cases could benefit from the `switch` syntax (e.g. more than one special type consideration, and in every case the operation just returns something):

```csharp
switch (source)
{
case IPartition<TSource> partition:
  return partition.Take(count);

case IList<TSource> sourceList:
  return new ListPartition<TSource>(sourceList, 0, count - 1);

default:
  return new EnumerablePartition<TSource>(source, 0, count - 1);
}
```